Automattic/wp-calypso
# Social First Signup: Better separation between User and UserSocial

## What

With https://github.com/Automattic/wp-calypso/pull/82481  we shipped the work done on pdDR7T-Z4-p2 .
We went from the usual signup step:
![image](https://github.com/Automattic/wp-calypso/assets/52076348/22253ce2-582d-41bb-8ab3-218cf466c53f)

To the new Social First Signup:
![image](https://github.com/Automattic/wp-calypso/assets/52076348/3e630509-afc7-4d32-be44-69febef3a769)

## The problem

Signup is a central and core part of the whole WordPress.com existence, and we shipped a change that required more time to be built, tested, and thought than we expected.

Following excellent suggestions on how to improve this, I worked on a better structure for this change.

## Solution

While the previous implementation just changed the User step and the SignupForm component, this creates a new step, `user-social`, that can or not be added to a flow. 
This new step calls the old User step but tells it to be `social-first`.

Moreover, we have a new flag, `signup/social-first`, that can be used to disable the new step globally if needed.

## Testing
1. Click the Live link
2. You can check, while in **Incognito**,  the flows that are in [this array here](https://github.com/Automattic/wp-calypso/blob/update/componentize-signup-social-first/client/signup/config/flows-pure.js). The ones that use `userSocialStep` have the new signup step.
3. Check that flows like `onboarding-pm` are not affected by the new signup. (Visit `/start/onboarding-pm` )
4. Check flows from Stepper, like `/setup/newsletter`, `/setup/link-in-bio` and `/setup/domain-transfer`.

2. While in incognito, go through these flows. They should have the standard signup:

- Gravatar `/log-in?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D500e79a3faed390d3ef5f2778ef39363706cc0609c02c3c25397c7499f3fa51c%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token`
![image](https://github.com/Automattic/wp-calypso/assets/52076348/5f1ccdf1-3062-4f99-8e6a-90dc33b9ffbd)
- P2: `/log-in?redirect_to=https%3A%2F%2Fwordpress.com%2Fstart%2Fp2%2Fp2-confirm-email%3Fref%3Dp2-lp&from=p2&signup_url=%2Fstart%2Fp2%2Fuser%3Fref%3Dp2-lp` 
![image](https://github.com/Automattic/wp-calypso/assets/52076348/e1f46369-1b1a-45d3-9228-4f913e4d328e)
-Crowdsignal: `/log-in?client_id=978&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D978%26response_type%3Dcode%26blog_id%3D0%26state%3D51d5e1b0d7358dee3de65c45f4b4dc37a3d6e37dab8d6047545f636c2659d104%26redirect_uri%3Dhttps%253A%252F%252Fapp.crowdsignal.com%252Fconnect%253Fsource%253Dlogin%2526action%253Drequest_access_token%26wpcom_connect%3D1`
![image](https://github.com/Automattic/wp-calypso/assets/52076348/35edd4cf-84a1-4203-b4a4-3cda1fa65869)
- Akismet: `/start/wpcc/oauth2-user?oauth2_client_id=973&oauth2_redirect=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D973%26response_type%3Dcode%26blog_id%3D0%26state%3D4a9b8708620f66f5e004f26f53a9678dab2d731a4a03f3bbb962415a3b0501cf%26redirect_uri%3Dhttps%253A%252F%252Fakismet.com%252Faccount%252F%253Faction%253Drequest_access_token%26implicit%3Dtrue%26new-user%3D0&user_email=`
![image](https://github.com/Automattic/wp-calypso/assets/52076348/7f068bd1-c595-4231-bbb5-c50bb68d5f44)
- Jetpack connect: `/log-in/jetpack?site=https%3A%2F%2Ffalse-shrike.jurassic.ninja&redirect_to=http%3A%2F%2Fcalypso.localhost%3A3000%2Fjetpack%2Fconnect%2Fauthorize%3Fclient_id%3D212713831%26redirect_uri%3Dhttps%253A%252F%252Ffalse-shrike.jurassic.ninja%252Fwp-admin%252Fadmin.php%253Fhandler%253Djetpack-connection-webhooks%2526action%253Dauthorize%2526_wpnonce%253D0e6a75dad7%2526redirect%253Dhttps%25253A%25252F%25252Ffalse-shrike.jurassic.ninja%25252Fwp-admin%25252Fadmin.php%25253Fpage%25253Djetpack%26state%3D1%26scope%3Dadministrator%253A3a6e1fb4a0d3fca03c4b16915582698f%26user_email%3Ddemo%2540example.org%26user_login%3Ddemo%26jp_version%3D11.5.1%26secret%3DvR0f4dhOxJWjtiika7pL3ns8z8JOsPHS%26blogname%3DFalse%2BShrike%26site_url%3Dhttps%253A%252F%252Ffalse-shrike.jurassic.ninja%26home_url%3Dhttps%253A%252F%252Ffalse-shrike.jurassic.ninja%26site_icon%26site_lang%3Den_US%26site_created%3D2022-11-23%2B12%253A55%253A29%26allow_site_connection%3D1%26locale%3Den%26_ui%3D335273%26_ut%3Dwpcom%253Auser_id%26_as%3Dwp-cli%26purchase_nonce%3DslDQZ1DY%26_wp_nonce%3D80f05dcb20%26redirect_after_auth%3Dhttps%253A%252F%252Ffalse-shrike.jurassic.ninja%252Fwp-admin%252Fadmin.php%253Fpage%253Djetpack%26site%3Dhttps%253A%252F%252Ffalse-shrike.jurassic.ninja&email_address=demo@example.org&from=%5Bunknown%5D&allow_site_connection=1`
